core/src/main/scala/io/chrisdavenport/shellfish/FilesOs.scala

```diff
@@ -705,14 +679,16 @@ object FilesOs {
   def size(path: Path): IO[Long] = files.size(path)
 
   /**
-   * Creates a temporary file and deletes it upon finalization of the returned
-   * resource.
+   * Creates a temporary file and deletes it at the end of the use of it.
    */
-  def tempFile: Resource[IO, Path] = files.tempFile(None, "", ".tmp", None)
+  def tempFile[A](use: Path => IO[A]): IO[A] =
```

Comment:
bike-shedding: I found `tempFile` name is a bit unintuitive. I suggest adding `use` prefix for all of these functions, which reflects how we implement it as well `files.tempFile.use(use)`

I'm still not very happy with my suggestion, I think I need to see some practical usage of it to have a better suggestion. Other suggestions are welcome.

```suggestion
  def useTempFile[A](use: Path => IO[A]): IO[A] =
```

Reply:
`withTempFile`?